lessons/cryptography/cryptosystem-types/cryptosystems-types.md

```diff
@@ -0,0 +1,90 @@
+
+### Diffie-Hellman
+The Diffie-Hellman key exchange process involves the use of asymmetric encryption, and is used to establish a shared private key for use in future communications. When we previously talked about AES (symmetric cryptosystem), we noted that Alice and Bob would need a way to securely share a private key. The Diffie-Hellman key exchange process can accomplish this. The Diffie-Hellman algorithm involves a public modulus and base value, and two secret values chosen by Alice and Bob. Through an elegant mathematical algorithm, Alice and Bob can each use their own private secret values to calculate a public key that can be sent to each other unencrypted. They can then each use the other person's public key and their own secret value to calculate the same secret key. 
+
+The strength of the Diffie-Hellman key exchange process depends on using a strong random number generator to choose the public modulus so that it cannot be easily predicted by an attacker. In addition, this process does not provide any method of authentication between Alice and Bob, so it is vulnerable to a man-in-the-middle attack where an attacker could intercept information between Alice and Bob and instead establish two separate Diffie-Hellman key exchanges with Alice and Bob. 
+
+### Elliptic Curve Cryptosystem
+Elliptic Curve Cryptosystems (ECC) are the most powerful type of cryptography currently used today. ECC is based on a mathematical concept called an elliptic curve, which is a set of points that satisfy a specific mathematical equation. For example, an elliptic curve is shown below. 
+
+![Sample Elliptic Curve (https://blog.cloudflare.com/a-relatively-easy-to-understand-primer-on-elliptic-curve-cryptography/)](.rsrc/elliptic-curve.png)
+
+ECC is based on solving a challenging mathematical problem that is considered more difficult to solve and more computationally expensive than the mathematical problem that RSA and Diffie-Hellman are based on. With ECC, smaller keys will achieve the same level of security compared to RSA. 
+
+## Symmetric vs Asymmetric Cryptography
+We previously discussed various symmetric and asymmetric cryptosystems. Below is a chart showing quick comparisons of some key features of both types of cryptosystems.
+
+| |Symmetric|Asymmetric|
```

Comment:
you can put spaces in these tables when you write them out so that they're easier to read in markdown

Reply:
Updated, but not sure if this is what you meant